src/nvim/version.c

```diff
@@ -57,7 +57,6 @@ static char *(features[]) = {
   "+cursorbind",
   "+cursorshape",
   "+dialog_con",
-  "+diff",
```

Comment:
This is solely for `--version` and `:version` output.

`has('feature')` is handled here https://github.com/neovim/neovim/blob/master/src/nvim/eval.c#L9892